api/gateway/state.go

```diff
@@ -149,16 +156,27 @@ func (h *Handler) handleSubmitPFB(w http.ResponseWriter, r *http.Request) {
 
 	// perform request
 	txResp, err := h.state.SubmitPayForBlob(r.Context(), fee, req.GasLimit, []*apptypes.Blob{blob})
-	if err != nil {
+	if err != nil && txResp == nil {
+		// no tx data to return
 		writeError(w, http.StatusInternalServerError, submitPFBEndpoint, err)
-		return
 	}
-	resp, err := json.Marshal(txResp)
+
+	resp := submitPFBResponse{
+		Tx:    txResp,
+		Error: err.Error(),
```

Comment:
We actually don't need to return an error so we can just return txResp. In SubmitPayForBlob we derive the error from the txResponse, so by returning the error we just send duplicate data.

Reply:
Ah, ok! TIL

Reply:
> Kinda weird API
💯

Reply:
Remved redundant error